tests/core/atoms/test_skzcam.py

```diff
@@ -1,146 +1,243 @@
-
-	# As these list of lists do not have the same length, we flatten first 5 lists into a 1D list for comparison
-	cation_shells_flatten = [item for row in cation_shells[:5] for item in row] 
-	cation_shells_idx_flatten = [item for row in cation_shells_idx[:5] for item in row]
-
-	# Check that these lists are correct
-	np.testing.assert_allclose(cation_shells_flatten,np.array([2.0,
- 3.6184221134101624,
- 3.6184221134101655,
- 3.6184221134101655,
- 3.6184221134101686,
- 4.646732760541734,
- 4.646732760541734,
- 4.646732760541736,
- 4.646732760541736,
- 4.6888354582307805,
- 4.6888354582307805,
- 4.6888354582307805,
- 4.6888354582307805,
- 6.267895285274443]),rtol=1e-05, atol=1e-07)
-	
-	assert np.all(cation_shells_idx_flatten == [0, 9, 8, 6, 7, 11, 12, 10, 13, 19, 21, 18, 20, 22])
+    # Get distance of atoms from the center
+    distances = _get_atom_distances(embedded_cluster, [0, 0, 2])
+
+    # Find the cation shells from the distances
+    cation_shells, cation_shells_idx = _find_cation_shells(embedded_cluster, distances)
+
+    # As these list of lists do not have the same length, we flatten first 5 lists into a 1D list for comparison
+    cation_shells_flatten = [item for row in cation_shells[:5] for item in row]
+    cation_shells_idx_flatten = [item for row in cation_shells_idx[:5] for item in row]
+
+    # Check that these lists are correct
+    np.testing.assert_allclose(
+        cation_shells_flatten,
+        np.array(
+            [
+                2.0,
+                3.6184221134101624,
+                3.6184221134101655,
+                3.6184221134101655,
+                3.6184221134101686,
+                4.646732760541734,
+                4.646732760541734,
+                4.646732760541736,
+                4.646732760541736,
+                4.6888354582307805,
+                4.6888354582307805,
+                4.6888354582307805,
+                4.6888354582307805,
+                6.267895285274443,
+            ]
+        ),
+        rtol=1e-05,
+        atol=1e-07,
+    )
+
+    assert np.all(
+        cation_shells_idx_flatten == [0, 9, 8, 6, 7, 11, 12, 10, 13, 19, 21, 18, 20, 22]
+    )
 
-def test_get_anion_coordination(embedded_cluster, distance_matrix):
 
-	# Get the anions for the second SKZCAM shell
-	anion_shell_idx = _get_anion_coordination(embedded_cluster, [9, 8, 6, 7], distance_matrix)
+def test_get_anion_coordination(embedded_cluster, distance_matrix):
+    # Get the anions for the second SKZCAM shell
+    anion_shell_idx = _get_anion_coordination(
+        embedded_cluster, [9, 8, 6, 7], distance_matrix
+    )
 
-	# Check anion indices match with reference
-	assert np.all(anion_shell_idx == [1, 2, 3, 4, 14, 15, 16, 17, 23, 24, 25, 26, 27, 28, 29, 30])
+    # Check anion indices match with reference
+    assert np.all(
+        anion_shell_idx == [1, 2, 3, 4, 14, 15, 16, 17, 23, 24, 25, 26, 27, 28, 29, 30]
+    )
```

Comment:
I think you mean to be using https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_equal.html here and throughout.